cmd/cli/metrics_enable.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+)
+
+const metricsEnableDescription = `
+This command will enable metrics on all pods belonging to the given namespace
+or set of namespaces. Newly created pods belonging to namespaces that
+are enabled for metrics will be automatically enabled with metrics.
+`
+
+type metricsEnableCmd struct {
+	out        io.Writer
+	namespaces []string
+	clientSet  kubernetes.Interface
+}
+
+func newMetricsEnable(out io.Writer) *cobra.Command {
+	enableCmd := &metricsEnableCmd{
+		out: out,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "enable ...",
+		Short: "enable metrics",
+		Long:  metricsEnableDescription,
+		Args:  cobra.NoArgs,
+		RunE: func(_ *cobra.Command, args []string) error {
+			config, err := settings.RESTClientGetter().ToRESTConfig()
+			if err != nil {
+				return errors.Errorf("Error fetching kubeconfig")
+			}
+
+			clientset, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				return errors.Errorf("Could not access Kubernetes cluster. Check kubeconfig")
+			}
+			enableCmd.clientSet = clientset
+			return enableCmd.run()
+		},
+	}
+
+	//add mesh name flag
+	f := cmd.Flags()
+	f.StringSliceVar(&enableCmd.namespaces, "namespace", []string{}, "One or more namespaces to enable metrics on")
+
+	return cmd
+}
+
+func (cmd *metricsEnableCmd) run() error {
```

Comment:
I think this would also have to check if [enablePrometheus](https://github.com/openservicemesh/osm/blob/main/charts/osm/values.yaml#L35) is enabled

Reply:
Or if we also support byo prometheus, we might have to check for that.

Reply:
Don't think its necessary, because these are just annotations being added. If the user doesn't deploy prometheus or disables prometheus scraping via osm-config configMap its expected for the metrics to not be scraped. Running the CLI just informs the controller the scope of scraping (namespace in this case).

Reply:
Sure, I think this is okay for now, but we should eventually consider returning some sort of message or at least document clearly so that the user understands that this could be a no-op if Prometheus is not deployed. In other words, we need to document clearly that Prometheus deployment is a pre-req because "enable metrics" is vague unless you know the implementation.

Reply:
This will go into the documentation that I am going to write as a part of #1016, which is why I am not going to mark the issue as resolved.

Reply:
Btw the command states that metrics will be enabled on pods, it doesn't say it will deploy prometheus. I could have changed the command to `osm metrics-collection` but it seemed longer than necessary. WDYT?

Reply:
Agree with Shashank. Adding the annotations will actually not enable any metrics collection until the prometheus scraping via osm-config is enabled (for either OSM Prometheus or BYO)

Reply:
Addressed in latest commit